enhancements/storage/storage-hypershift.md

```diff
@@ -0,0 +1,973 @@
+---
+title: storage-hypershift
+authors:
+- "@jsafrane"
+reviewers:
+- "@gnufied"
+- "@dobsonj"
+- "@csrwng"
+approvers:
+- "@derekwaynecarr"
+api-approvers:
+ - None
+creation-date: 2022-07-26
+last-updated: 2022-07-27
+tracking-link:
+- https://issues.redhat.com/browse/STOR-962
+see-also:
+replaces:
+superseded-by:
+---
+
+# Running storage operators in HyperShift management cluster
+
+## Summary
+
+This enhancement describes how OCP storage components will run in HyperShift's managed control plane and in the guest
+cluster.
+
+The components currently are:
+
+* cluster-csi-snapshot-controller-operator and its operand, csi-snapshot-controller.
+* cluster-storage operator and its operand aws-ebs-csi-driver-operator and its operand AWS EBS CSI driver.
+  * All other CSI driver operators + CSI drivers should follow the same approach, AWS EBS is just the first one to
+    run its control plane in HyperShift's management cluster.
+  
+## Motivation
+
+A guest cluster in HyperShift should ideally run only user workloads, and it should not be possible to break the cluster
+from within it. In this enhancement, we're getting closer to this goal, however, an evil cluster-admin can still break
+storage parts of the guest cluster. See Risks and Mitigations below.
+
+### User Stories
+
+* As a guest cluster user, I want the cluster to run as little OCP infrastructure workloads as possible to have the
+  cluster just for my applications.
+  
+* As a guest cluster user, I want the cluster to be as robust as possible, I should not be able to break it
+  by my actions.
+
+* As a guest cluster user, I don't want OCP infrastructure components running there to have elevated RBACs, so
+  a compromised component cannot break my cluster (too much).
+  * Still, CSI driver Pods *must* run on each worker node in the guest cluster as privileged, to be able to format and
+    mount volumes. A compromised CSI driver Pod can do anything to the application data - steal it, delete it, encrypt
+    it..., and it can fully compromise the node.
+    
+* As a HyperShift user, I want to spawn new OCP guest clusters on AWS, so the guest clusters can use AWS EBS storage
```

Comment:
A "hypershift user" has no concrete meaning, please use one of the personas defined here https://hypershift-docs.netlify.app/reference/concepts-and-personas/

Reply:
Changed `user` everywhere to a proper role from the linked doc + linked the doc.